yarn-project/end-to-end/src/e2e_state_vars.test.ts

```diff
@@ -288,5 +289,66 @@ describe('e2e_state_vars', () => {
     });
+
+    it('checks max block number propagated by reading with our SharedMutablePrivateGetter in TestContract accessing AuthContract; before and after a decreasing then increasing delay change', async () => {
+      // Our initial max block number will be 5 because that is the value of INITIAL_DELAY
+      const expectedInitialMaxBlockNumber = (await pxe.getBlockNumber()) + 5;
+
+      // Our SharedMutablePrivateGetter here reads from the SharedMutable authorized storage property in AuthContract
+      const tx = await testContract.methods.test_shared_mutable_private_getter(authContract.address, 2).prove();
+
+      // The validity of our SharedMutable read request is limited to 5 blocks, which is our initial delay.
+      expect(tx.data.forRollup!.rollupValidationRequests.maxBlockNumber.isSome).toEqual(true);
+      expect(tx.data.forRollup!.rollupValidationRequests.maxBlockNumber.value).toEqual(
+        new Fr(expectedInitialMaxBlockNumber),
+      );
+
+      // We change the SharedMutable authorized delay here to 100, this means that a change to the "authorized" value can
+      // only be applied 100 blocks after it is initiated, and thus read requests on a historical state without an initiated change is
+      // valid for at least 100 blocks
+      await authContract.methods.set_authorized_delay(2).send().wait();
+
+      await delay(4);
+
+      const expectedModifiedMaxBlockNumber = (await pxe.getBlockNumber()) + 2;
+
+      // We now call our AuthContract to see if the change in max block number has reflected our delay change
+      const tx2 = await authContract.methods.get_authorized_in_private().prove();
+
+      // The validity of our SharedMutable read request should now be limited to 100 blocks, instead of 5
+      expect(tx2.data.forRollup!.rollupValidationRequests.maxBlockNumber.isSome).toEqual(true);
+      expect(tx2.data.forRollup!.rollupValidationRequests.maxBlockNumber.value).toEqual(
+        new Fr(expectedModifiedMaxBlockNumber),
+      );
+
+      // We access the same SharedMutable from our SharedMutablePrivateGetter, expecting the validity assumptions to remain the same.
+      const tx3 = await testContract.methods.test_shared_mutable_private_getter(authContract.address, 2).prove();
+
+      expect(tx3.data.forRollup!.rollupValidationRequests.maxBlockNumber.isSome).toEqual(true);
+      expect(tx3.data.forRollup!.rollupValidationRequests.maxBlockNumber.value).toEqual(
+        new Fr(expectedModifiedMaxBlockNumber),
+      );
+
+      await authContract.methods.set_authorized_delay(100).send().wait();
+
+      const expectedLengthenedModifiedMaxBlockNumber = (await pxe.getBlockNumber()) + 100;
+
+      // We now call our AuthContract to see if the change in max block number has reflected our delay change
+      const tx4 = await authContract.methods.get_authorized_in_private().prove();
+
+      // The validity of our SharedMutable read request should now be limited to 100 blocks, instead of 5
```

Comment:
Stale comment. 

This one is always a little funky to me. 
So you reduced it to 2, and now you are increasing it 100. 

The reduction needs to wait for the full execution before it is "happy", but for the extension it can take effect more quickly where there is not a pending value change 🤔